### ISSUE TITLE
modparser error when using a placeholder/element as array key in form input name.

### ISSUE THREAD

**Issue by JoshuaLuckers** [modxcms/revolution#13903]
### Summary
modparser gives an error and produces incorrect HTML when you want to use a placeholder or output of an element as array key in a form input name.

### Step to reproduce
Use the HTML snippet below as content in a page, chunk or template:
```html
<input type="text" name="items[[[*id]]][foo]" value="something">
<input type="text" name="items[[[*id]]][bar]" value="another thing">
```

### Observed behavior
The output modParser generates is wrong:
```html
<input type="text" name="items][foo]" value="something">
<input type="text" name="items][bar]" value="another thing">
```

An error is also written to the log:
```
ERROR @ core/model/modx/modparser.class.php : 540) Could not find snippet with name [*id.
```

### Expected behavior
Correct HTML:
```html
<input type="text" name="items[1][foo]" value="something">
<input type="text" name="items[1][bar]" value="another thing">
```

### Environment
MODX 2.6.3-pl

**Comment by Jako** [comment]
Since MODX Tags can't be escaped: Could you try to use `[ [[*id]] ][foo]`?

**Comment by Jako** [comment]
or use ```[[*id:before=`[`:after=`]`]]```

**Comment by JoshuaLuckers** [comment]
@Jako I made a pull request that solves the issue in my use case. I'm not sure if it will cause any issues in other use cases.

**Comment by JoshuaLuckers** [comment]
This is fixed by merging PR #13904
